index.js

```diff
@@ -72,6 +72,23 @@ module.exports = {
     "simple-import-sort/imports": "error",
     "simple-import-sort/exports": "error",
     "sort-destructure-keys/sort-destructure-keys": "error",
+    /**
+     * List of sonarjs rules we turned off after release of eslint-plugin-sonarjs v2
+     * Some rules are too restrictive for common use cases,
+     * and others are already covered by other plugins used here.
+     */
+    "sonarjs/different-types-comparison": "off",
+    "sonarjs/function-return-type": "off",
+    "sonarjs/no-clear-text-protocols": "warn",
+    "sonarjs/no-ignored-exceptions": "off",
+    "sonarjs/no-misused-promises": "off",
+    "sonarjs/no-nested-functions": "off",
+    "sonarjs/prefer-nullish-coalescing": "off",
+    "sonarjs/sonar-no-unused-vars": "off",
+    "sonarjs/sonar-prefer-optional-chain": "off",
```

Comment:
Useful for condensing conditional logic, which is a common code comment in PR reviews:

```diff
- contextMenuItems && contextMenuItems.length
+ contextMenuItems?.length
```

This is, however, also a `@typescript-eslint` rule: https://typescript-eslint.io/rules/prefer-optional-chain/. So regardless of `error` or `warn`, we'd want to enforce it through that, leaving this `off` comment as:

`// Overlaps with @typescript-eslint/prefer-optional-chain`.